### DISCUSSION TITLE
`stale` stuck at true after mutation?

### DISCUSSION THREAD

**Discussion by EdNutting** [urql-graphql/urql#2289] [Q&A]
Hi!

We have two queries:

1. Select Item

    `query SelectItem($id: uuid!) { content_Item_by_pk(id: $id) { ..., elements { ...ElementFragment }, ... }`

    This selects a whole bunch of data about an "item", including the "elements" of the item
2. Add Element

    `mutation InsertElement(...) { insert_content_Element_one(object: ...) { ...ElementFragment }`

    Inserts a new element for the item fetched by the outer Select Item query.

So we select an item, extract the elements, display them in a list, and have a button to add a new element to the list. When someone clicks the to add an element, we execute the insert mutation (and update the cache), and separately we use the refetch function returned by useQuery to refetch the entire item (due to possible backend side effects on the item).

We have a wrapper around our `useSelectItemQuery` call that abstracts common error handling etc. We recently added a condition that shows a loading spinner while `stale: true` because we found that the UI flickered too much when inserting a new element for an item. The UI would bounce around a bit while the insert mutation and cache updates were happening. We expected `stale` would enable us to just wait for when the refetched item query came back with the final state.

Unfortunately, we're getting stuck on `stale: true` permanently. And I mean like, really permanently. We do the initial select item (no problem here, `stale: false`), we execute the insert mutation (ok, `stale: true` as expected now) and then execute the refetch - `stale` gets stuck as `true` and will never return to `false`. We can navigate to entire other pages (i.e. entirely terminate any dependency on any of these queries/mutations previously executed) and then renavigate to the page with the select item. It reexecutes the select item query as a cache-and-network request, the result comes back from the network and `stale` is still `true`.

In effect, after executing the insert element mutation, there is no way back for that query. It will forever be stuck on `stale: true`. However, it will be `stale: false` for the same query but with different variables (i.e. for different items). Toggling back and forth between items, those with mutations previously executed are stuck at `stale: true` and ones without mutations are not.

Additional context:

* We use Graphcache for normalized caching
* We have sophisticated custom resolvers and updaters - if there is something in Graphcache that affects the behaviour of stale we probably don't know about it so maybe we're doing something wrong that's interfering?
* There are no key/configuration or network errors at any point

Thanks for your assistance!
Ed

P.s. Originally I posted this to an issue that wasn't related to this and immediately realised my error, so deleted the comment and am reposting as a separate discussion here.

Edit: Also pinging @rossng who has been looking into this too.

**Comment by gustavlrsn** [comment]
@EdNutting did you get any clarity on this? Also experiencing `stale` stuck to `true` (trying to set up `staleWhileRevalidated: true`)

**Reply by EdNutting** [reply]
Sorry, Ross and I never got anywhere with it. Time pressure meant we had to move onto other things and abandon using the `stale` flag.

**Reply by JoviDeCroock** [reply]
Hmm, I just got to this. It's hard to just point at what the issue is, if either of you has a reproduction I'd love to take a closer look 😅 

This is a sandbox that kinda does what you describe in the issue https://codesandbox.io/s/wizardly-panna-y2moe
